test/integration/features/features_test.go

```diff
@@ -0,0 +1,124 @@
+package features
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/nginx/agent/test/integration/utils"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/compose"
+	wait "github.com/testcontainers/testcontainers-go/wait"
+)
+
+func SetupTestContainerWithAgent(t *testing.T, conf string, waitForLog string) *testcontainers.DockerContainer {
+	comp, err := compose.NewDockerCompose(os.Getenv("DOCKER_COMPOSE_FILE"))
+	assert.NoError(t, err, "NewDockerComposeAPI()")
+
+	ctx := context.Background()
+	t.Cleanup(func() {
+		assert.NoError(t, comp.Down(ctx, compose.RemoveOrphans(true), compose.RemoveImagesLocal), "compose.Down()")
+	})
+
+	ctxCancel, cancel := context.WithCancel(ctx)
+	t.Cleanup(cancel)
+
+	require.NoError(t,
+		comp.WaitForService("agent", wait.ForLog(waitForLog)).WithEnv(
+			map[string]string{
+				"PACKAGE_NAME":  os.Getenv("PACKAGE_NAME"),
+				"PACKAGES_REPO": os.Getenv("PACKAGES_REPO"),
+				"BASE_IMAGE":    os.Getenv("BASE_IMAGE"),
+				"OS_RELEASE":    os.Getenv("OS_RELEASE"),
+				"OS_VERSION":    os.Getenv("OS_VERSION"),
+				"CONF_FILE":     conf,
+			},
+		).Up(ctxCancel, compose.Wait(true)), "compose.Up()")
+
+	testContainer, err := comp.ServiceContainer(ctxCancel, "agent")
+	require.NoError(t, err)
```

Comment:
Can you add something like this as well to capture the container logs? https://github.com/nginx/agent/pull/365/files#diff-a50d8e8f491f3806396a87e9bb64fdbe13160d438747a610b16a1f48d6fed2e1R43

You will probably need to update the ci.yaml as well like this https://github.com/nginx/agent/pull/365/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR122